### ISSUE TITLE
Bug: SVG component fails to render when containing CDATA

### ISSUE THREAD

**Issue by mfulton26** [facebook/react#22518]
React version:

## Steps To Reproduce

1. Create an SVG file
2. Place some CDATA node in it (e.g. inside a script tag)
3. Confirm it loads find in a web browser (and that the CDATA code if placed inside a `<script>` tag executes as expected, although the CDATA could be used for other things other than code in SVG files)
4. Use the SVG file as a ReactComponent in a React app (e.g. via CRA)

Link to code example:

https://codesandbox.io/s/svgmaterialicons-material-demo-forked-2l9hp?file=/hello.svg

## The current behavior

fails to render

```console
Error
Expected node, got `console.log("Hello")`
```

## The expected behavior

renders the SVG inside the app the same as loading the SVG directly

**Comment by lubieowoce** [comment]
Based on the stack trace, this looks like a problem with the svg library that some component is using under the hood, not a React issue. Check out this issue on their side though: https://github.com/gregberge/svgr/issues/558

**Comment by mfulton26** [comment]
I created the SVG by hand and didn't use a library. React scripts makes it into a component so React itself must be pulling in the library, right?

**Comment by lubieowoce** [comment]
yeah, i'd guess `react-scripts` (not React itself) uses this webpack plugin ([@svgr/webpack](https://react-svgr.com/docs/webpack/)) that turns the SVG into JSX/a component. unfortunately i don't think there's anything that can be done here until `@svgr` supports CDATA.

you may be able to work around svgr by importing the SVG as text somehow (without svgr turning it into JSX) and manually using `dangerouslySetInnerHtml` as described here:  https://github.com/facebook/react/issues/2250#issuecomment-229178783

**Comment by mfulton26** [comment]
I suppose then at a minimum this issue can be closed and one can be opened in https://github.com/facebook/create-react-app but even then the issue may need to be fixed downstream

Thank you for your assistance

**Comment by azizur** [comment]
Looks like a fix is available in https://github.com/gregberge/svgr/issues/558#issuecomment-1063557878

**Comment by eps1lon** [comment]
Closing in favor of https://github.com/gregberge/svgr/issues/558#issuecomment-1063557878 since this is not caused by React itself.
